index.js

```diff
@@ -56,85 +71,142 @@ export type Options = {
+|};
+
+export type UserCredentials = {|
+  +username: string,
+  +password: string,
+  ...Result,
+|};
+
+export type SharedWebCredentials = {|
+  +server: string,
+  ...UserCredentials,
+|};
+
+function getOptionsArgument(serviceOrOptions?: string | Options) {
+  if (Platform.OS !== 'ios') {
+    return typeof serviceOrOptions === 'object'
+      ? serviceOrOptions.service
+      : serviceOrOptions;
+  }
+  return typeof serviceOrOptions === 'string'
+    ? { service: serviceOrOptions }
+    : serviceOrOptions;
+}
+
+function getAccessControl(serviceOrOptions?: string | Options) {
+  var accessControl = null;
+
+  if (typeof serviceOrOptions === 'object') {
+    accessControl = serviceOrOptions.accessControl;
+  }
+
+  return accessControl;
+}
+
+function getMinimumSecurityLevel(serviceOrOptions?: string | Options) {
+  var specifiedLevel = undefined;
+
+  if (typeof serviceOrOptions === 'object') {
+    specifiedLevel = serviceOrOptions.securityLevel;
+  }
+
+  return specifiedLevel || SECURITY_LEVEL.ANY;
+}
+
 /**
- * (Android only) Returns guaranteed security level supported by this library
- * on the current device.
- * @return {Promise} Resolves to `SECURITY_LEVEL` when supported, otherwise `null`.
+ * Saves the `username` and `password` combination for `service`.
+ * @param {string} username Associated username or e-mail to be saved.
+ * @param {string} password Associated password to be saved.
+ * @param {object} options An Keychain options object.
+ * @return {Promise} Resolves to `{ service, storage }` (for Android) or `true` (for iOS) when successful
  */
-export function getSecurityLevel(): Promise<?($Values<typeof SECURITY_LEVEL>)> {
-    if (!RNKeychainManager.getSecurityLevel){
-        return Promise.resolve(null);
-    }
-    return RNKeychainManager.getSecurityLevel();
+export function setGenericPassword(
+  username: string,
+  password: string,
+  options?: Options
+): Promise<boolean | Result> {
+  var obj: { service?: string } = options || {};
+
+  return RNKeychainManager.setGenericPasswordForOptions(
+    obj.service,
+    username,
+    password,
+    options
+  );
 }
 
 /**
- * Inquire if the type of local authentication policy (LAPolicy) is supported
- * on this device with the device settings the user chose.
- * @param {object} options LAPolicy option, iOS only
- * @return {Promise} Resolves to `true` when supported, otherwise `false`
+ * Fetches login combination for `service`.
+ * @param {object} options An Keychain options object.
+ * @return {Promise} Resolves to `{ service, username, password, storage }` when successful
  */
-export function canImplyAuthentication(options?: Options): Promise<boolean> {
-  if (!RNKeychainManager.canCheckAuthentication) {
-    return Promise.resolve(false);
-  }
-  return RNKeychainManager.canCheckAuthentication(options);
+export function getGenericPassword(
+  options?: Options
+): Promise<boolean | SharedWebCredentials> {
+  var obj: { service?: string } = options || {};
```

Comment:
Why change this from using `getOptionsArgument` like before?

Reply:
To make all API identical in usage pattern